cpp/include/cudf/strings/detail/gather.cuh

```diff
@@ -85,10 +85,10 @@ CUDF_KERNEL void gather_chars_fn_string_parallel(StringIterator strings_begin,
   constexpr size_t out_datatype_size = sizeof(uint4);
   constexpr size_t in_datatype_size  = sizeof(uint);
 
-  int global_thread_id = blockIdx.x * blockDim.x + threadIdx.x;
+  int global_thread_id = cudf::detail::grid_1d::global_thread_id();
```

Comment:
I think we want to change this to `auto` or `thread_index_type`.

Reply:
Oopsie. Yes, thanks.